config/initializers/assets.rb

```diff
@@ -8,7 +8,7 @@
 
 # Precompile additional assets.
 # application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
-Rails.application.config.assets.precompile += %w(misc/*.js email.css ie*.css)
+Rails.application.config.assets.precompile += %w(app/controllers/*.js misc/*.js email.css ie*.css)
```

Comment:
So it seems like this file is not getting target for compilation unless I add it to the application.js file, which means it tries to run on every page. The original reason I had it as a helper method was so this JS was only loaded on the screens that required it.

Seems like adding it manually to the asset compilation array also achieved that